### ISSUE TITLE
Branches not being tracked after removing jar file

### ISSUE THREAD

**Issue by Teddayz** [nus-cs2103-AY2425S1/forum#366]
I referred to the bot feedback and it told me to remove my jar files from my remote repository. 
Initially, I created a randomFolder to test the jar file but I forgot to add it to the .gitignore.

I ran the command git filter-repo --path randomFolder/data/meeks.jar --invert-paths
This removed all the meeks.jar files from all the previous commits that I had.

However, when I refreshed the IP Dashboard this morning, all my branches are now red instead of green.
![image](https://github.com/user-attachments/assets/fd5c512d-9983-46e5-9c44-b3aef568bf73)
Furthermore, I didn't cant go back to the previous version already and did not save into another repository.

![image](https://github.com/user-attachments/assets/64e43f21-dcad-4468-b034-3701bda1653b)
I have my remote branches still on github

Prior to this, all my branches were correctly tracked. Will I be marked down for the IP my branches are still red?
How do I proceed?

**Comment by damithc** [comment]
@Teddayz Looks like something went wrong with your branches.
Not to worry. We should be able to figure this out and rectify in due course.
At the top of the [iP progress dashboard](https://nus-cs2103-ay2425s1.github.io/dashboards/contents/ip-progress.html), there are a couple of tips about troubleshooting when an increment related to a tag or a branch is not green as expected. See if those help you locate the problem.
You can also post your git version graph here so that others can get a better context.

**Comment by Teddayz** [comment]
Hello, this is my version graph when I type in git log --graph --oneline

![image](https://github.com/user-attachments/assets/69ed6261-cd5f-49f4-a29e-b24b779b6958)

Referring to the tips for troubleshooting, my current branches look like this. I have went to some individual branches and pulled from the remote repository. Then, I removed the jar file before committing and pushing back to the branch.
![image](https://github.com/user-attachments/assets/47377f14-ee7c-497c-90e7-7420e563fabf)
Im afraid to merge it to my current master branch as it might mess up my finished IP.

**Comment by damithc** [comment]
> I ran the command git filter-repo --path randomFolder/data/meeks.jar --invert-paths
> This removed all the meeks.jar files from all the previous commits that I had.

In other words, all those commits became new commits, different from the commits in your fork. Hence, the version of the branch in GitHub is no longer considered as merged to your master (that is why the `Ahead` number in the second screenshot is not zero for those branches, as mentioned in the troubleshooting guide)

**Comment by damithc** [comment]
One option is to _force_ the branches in the fork to follow your local corresponding branch.
For each branch, you can `git push -f origin BRANCH-NAME` e.g., `git push -f origin branch-A-UserGuide` to overwrite the branch in the fork with the one in the local repo. You can try with one branch first, and wait to see if turns green after the next dashboard update.

**Comment by Teddayz** [comment]
Could I merge each branch to my master branch and merge the conflicts as well?

**Comment by Teddayz** [comment]
I have went to each individual branch and ensured that they are the latest version before pushing the branch again to the remote repository. Then I went to master branch and merged them 1 by 1 whilst resolving the merge conflicts. This ensured that all my branches are merged to my master branch. 

![image](https://github.com/user-attachments/assets/b596df3f-afd2-4598-8298-5348ac9018b1)
Now my commits have the second number as 0.

Does this solve the problem? Waiting to check for the IP dashboard tomorrow.
